jdaviz/configs/imviz/plugins/rotate_image/rotate_image.py

```diff
@@ -0,0 +1,35 @@
+import math
+
+from traitlets import Any
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import TemplateMixin, ViewerSelectMixin
+
+
+@tray_registry('imviz-rotate-image', label="Simple Image Rotation")
+class RotateImageSimple(TemplateMixin, ViewerSelectMixin):
+    template_file = __file__, "rotate_image.vue"
+
+    angle = Any(0).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._theta = 0  # degrees, clockwise
+
+    def vue_rotate_image(self, *args, **kwargs):
+        # We only grab the value here to avoid constantly updating as
+        # user is still entering or updating the value.
+        try:
+            self._theta = float(self.angle)
+        except Exception:
+            return
+
+        viewer = self.app._viewer_by_id(self.viewer_selected)
+
+        # Rotate selected viewer canvas.
+        # TODO: Translation still a bit broken if zoomed in.
+        viewer.state.rotation = math.radians(self._theta)
+
+        # TODO: Zoom box in Compass still broken, how to fix? If not, we need to disable it.
+        # Update Compass plugin.
+        viewer.on_limits_change()

```

Comment:
How do I get viewer limits that is consistent with the rotation, so I can show a sane zoom box in Compass plugin when image is rotated?

Reply:
Are you going to show the image rotated in the compass plugin as well? Or are you going to show it in the original/native orientation and want to show the box rotated? (since it matters for the limits)

Reply:
I was thinking it is easier in Compass if we do not rotate the image but only rotate the zoom box. The Compass display real estate is already pretty small. To rotate the image in Compass would make the image display even smaller when rotated.

Also even if not rotated, images can have wild orientations after you link them by WCS.